mpc/coordinator/server_configuration.py

```diff
@@ -0,0 +1,128 @@
+            email_password: Optional[str] = None):
+        if not contributors_file:
+            raise Exception("no contributors file specified")
+        if 0 == start_time:
+            raise Exception("invalid start time")
+        if (email_server or email_address or email_password) and \
+           (not (email_server and email_address and email_password)):
+            raise Exception(
+                "must all or none of email server, address and password " +
+                "in config")
+
+        self.contributors_file: str = contributors_file
+        self.start_time: float = float(start_time)
+        self.contribution_interval: float = float(contribution_interval)
+        self.email_server: Optional[str] = email_server
+        self.email_address: Optional[str] = email_address
+        self.email_password: Optional[str] = email_password
+        self.tls_key: str = tls_key
+        self.tls_certificate: str = tls_certificate
+        self.port = port
+
+    @staticmethod
+    def template() -> Configuration:
+        """
+        Populate contributors field, and other fields with sensible defaults
+        for a configuration template.  All fields are expected to be
+        overridden.
+        """
+        return Configuration(
+            contributors_file="contributors.json",
+            start_time=time.time() + 6 * 60 * 60,
+            contribution_interval=24 * 60 * 60,
+            tls_key="key.pem",
+            tls_certificate="cert.pem",
+            port=5000,
+            email_server="smtp.mymail.com:465",
+            email_address="mpc_coordinator@mymail.com",
+            email_password="*")
+
+    def to_json(self) -> str:
+        return json.dumps(self._to_json_dict(), indent=4)
+
+    def to_json_template(self) -> str:
+        """
+        For the case where ana administrator has a list of contributors (e.g.
+        from an online form) and keys and wants to import it into a
+        server_config.json file. This function writes the contributors
+        correctly, and places dummy data / descriptions in other fields,
+        prefixed with '_', for the admin to fill in later.
+        """
+        return json.dumps(self._to_json_template_dict(), indent=4)
+
+    @staticmethod
+    def from_json(config_json: str) -> Configuration:
+        return Configuration._from_json_dict(json.loads(config_json))
+
+    def _to_json_dict(self) -> JsonDict:
+        start_local = time.localtime(self.start_time)
+        return {
+            "contributors_file": self.contributors_file,
+            "start_time": time.strftime(TIME_FORMAT, start_local),
+            "contribution_interval": str(self.contribution_interval),
+            "email_server": self.email_server,
+            "email_address": self.email_address,
+            "email_password": self.email_password,
+            "tls_key": self.tls_key,
+            "tls_certificate": self.tls_certificate,
+            "port": self.port,
+        }
```

Comment:
Don't you need to account for the path of the wrapped binaries here, @dtebbs ?

Reply:
This is the common server settings type.  There should be a new type, one for each phase, defining the config for that phase (which will include this common config as a "server" attribute).
I considered using inheritance (so `Phase1Config` inherits from `Configuration`), but thought it would be neater to have the common parts grouped rather than mixed in with the phase-specific parts.

Reply:
Ah right, thanks for the quick answer